sympy/solvers/diophantine/diophantine.py

```diff
@@ -190,27 +188,391 @@ def matches(self):
         """
         return False
 
+    def solve(self, params=None, limit=None) -> DiophantineSolutionSet:
+        raise NotImplementedError('No solver has been written for %s.' % self.name)
+
 
 class Univariate(DiophantineEquationType):
     name = 'univariate'
 
     def matches(self):
         return self.dimension == 1
 
+    def solve(self, params=None, limit=None):
+        result = DiophantineSolutionSet(self.free_symbols, parameters=params)
+        for i in solveset_real(self.equation, self.free_symbols[0]).intersect(S.Integers):
+            result.add((i,))
+        return result
+
 
 class Linear(DiophantineEquationType):
     name = 'linear'
 
     def matches(self):
         return self.total_degree == 1
 
+    def solve(self, params=None, limit=None):
+        """
+        Solves diophantine equations of the form:
```

Comment:
Would this make more sense as a class-level docstring?

Reply:
Yes. I will merge this with the docstring that is currently in the `diop_linear` function.